docs/test-prep.md

```diff
@@ -0,0 +1,406 @@
+
+Data entities are important because this is where a `suffix` or `prefix` can be defined. This ensures that data values can be unique for every test run.
+
+Notice that the `<entity>` name is `_defaultProduct` as referenced above. Within this entity is the `name` value.
+
+```xml
+<entities xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:DataGenerator/etc/dataProfileSchema.xsd">
+    <entity name="_defaultProduct" type="product">
+        <data key="sku" unique="suffix">testSku</data>
+        <data key="type_id">simple</data>
+        <data key="attribute_set_id">4</data>
+        <data key="visibility">4</data>
+   ---> <data key="name" unique="suffix">testProductName</data>
+        <data key="price">123.00</data>
+        <data key="urlKey" unique="suffix">testurlkey</data>
+        <data key="status">1</data>
+        <data key="quantity">100</data>
+        <data key="weight">1</data>
+        <requiredEntity type="product_extension_attribute">EavStockItem</requiredEntity>
+        <requiredEntity type="custom_attribute_array">CustomAttributeCategoryIds</requiredEntity>
+    </entity>
+```
+
+The `unique="suffix"` attribute appends a random numeric string to the end of the actual data string. This ensures that unique values are used for each test run.
+
+XXXX for difficult data needs, see https://devdocs.magento.com/mftf/docs/data.html
+
+## Convert actions to action groups
+
+Action groups are sets of steps that are run together. Action groups are designed to break up multiple individual steps into logical groups. For example: Logging into the admin panel requires ensuring the login form exists, filling in 2 form fields and clicking the Submit button. These can be bundled into a single, reusable "Admin Login" action group that can be applied to any other test. This leverages existing code and prevents duplication of effort. We recommend that all steps in a test be within an action group.
+
+Using action groups can be very useful when testing extensions.
+Extending the example above, assume the first extension adds a new field to the admin login, a Captcha for example.
+The second extension we are testing needs to login AND get past the Captcha.
+
+1. The admin login is encapsulated in an action group.
+2. The Captcha extension properly extends the `admin login` capture group using the `merge` functionality.
+3. Now the second extension can call the `admin login` action group and because of the `merge`, it will automatically include the Captcha field.
+
+In this case, the action group is both reusable and extensible!
+
+We further abstract the test by putting these action groups in their own file: ['app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminProductActionGroup.xml'](https://raw.githubusercontent.com/magento-pangolin/magento2/e5671d84aa63cad772fbba757005b3d89ddb79d9/app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminProductActionGroup.xml)
```

Comment:
the link points to `magento-pangolin` organization
Consider changing it to point to the `magento` organization, and to point to the latest released code like https://github.com/magento/magento2/blob/2.3/app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminProductActionGroup.xml

Reply:
That part lives only under Pangoling org repo. And this branch was created only to demonstrate how it should work. 
So maybe we should include it into document or those branches forever.

Reply:
We decided to leave the link for now.